src/ziggurat/middleware/default.clj

```diff
@@ -35,5 +64,7 @@
 (defn protobuf->hash
   "This is a middleware function that takes in a message (Proto ByteArray or PersistentHashMap) and calls the handler-fn with the deserialized PersistentHashMap"
   [handler-fn proto-class topic-entity-name]
-  (fn [message]
-    (handler-fn (deserialise-message message proto-class topic-entity-name))))
+  (if (config/get-in-config [:alpha-features :protobuf-middleware :enabled])
```

Comment:
since the two functions have same parameter/args, we can just have this function returned from another function, something like this
```(get-deserialize-message-fn []
(if (config/get-in-config [:alpha-features :protobuf-middleware :enabled])
  deserialise-message
  deserialise-message-deprecated))
```
and just call the returned function from here

Reply:
But we are executing the handler function inside the clojure

Reply:
We are using this condition to identify which function to add to the middleware stack and to make the code readable, we made it this way